arch/riscv/core/pmp.c

```diff
@@ -207,7 +207,8 @@ static bool set_pmp_entry(unsigned int *index_p, uint8_t perm,
  */
 extern void z_riscv_write_pmp_entries(unsigned int start, unsigned int end,
 				      bool clear_trailing_entries,
-				      ulong_t *pmp_addr, ulong_t *pmp_cfg);
+				      const ulong_t *pmp_addr,
+				      const ulong_t *pmp_cfg);
```

Comment:
I wonder how we maintain these workarounds, or better, how do we get rid of them? Once zephyr sdk gets a version of qemu with the fix?

Reply:
Either that or the Zephyr SDK incorporates the workaround in its QEMU version. Given the workaround is easy and functionality isn't impacted, I'd wait until the SDK gets a fixed QEMU and then revert those workarounds. In any case I'm keeping a virtual eye on it.